tokio-trace-fmt/src/time.rs

```diff
@@ -0,0 +1,110 @@
+#[cfg(feature = "chrono")]
+use chrono;
+
+#[cfg(feature = "ansi")]
+use ansi_term::Style;
+
+use std::fmt;
+use std::time::Instant;
+
+/// A type that can measure and format the current time.
+///
+/// This trait is used by `Format` to include a timestamp with each `Event` when it is logged.
+///
+/// Notable default implementations of this trait are `SystemTime` and `()`. The former prints the
+/// current time as reported by `std::time::SystemTime`, and the latter does not print the current
+/// time at all. `FormatTime` is also automatically implemented for any function pointer with the
+/// appropriate signature.
+pub trait FormatTime {
+    /// Measure and write out the current time.
+    ///
+    /// When `format_time` is called, implementors should get the current time using their desired
+    /// mechanism, and write it out to the given `fmt::Write`. Implementors must insert a trailing
+    /// space themselves if they wish to separate the time from subsequent log message text.
```

Comment:
👍 for documenting the contract around whitespace. I'm not convinced this should be the `FormatTime` impl's responsibility, though? Not a big deal & I could go either way, as long as it's documented.

Reply:
So, the reason it is is so that `()` can implement `FormatTime` and thus be used as the timer for `Format` when timestamps should not be included. If this were not the case, we'd need to store an `Option<FormatTime>` inside `Format` and then optionally write out the whitespace. We _could_ do that, but this avoids that branch and the associated `Option` boilerplate.